lib/dotdiff/element_handler.rb

```diff
@@ -9,27 +9,29 @@ def initialize(driver, elements = DotDiff.xpath_elements_to_hide)
 
     def hide
       elements.each do |xpath|
-        if element_exists?(xpath)
-          driver.execute_script("#{js_element(xpath)}.style.visibility = 'hidden'")
-        end
+        driver.execute_script(script(xpath, 'hidden')) if elements_exists?(xpath)
       end
     end
 
     def show
       elements.each do |xpath|
-        if element_exists?(xpath)
-          driver.execute_script("#{js_element(xpath)}.style.visibility = ''")
-        end
+        driver.execute_script(script(xpath, '')) if elements_exists?(xpath)
       end
     end
 
-    def js_element(xpath)
-      "document.evaluate(\"#{xpath}\", "\
-      "document, null, XPathResult.FIRST_ORDERED_NODE_TYPE, null).singleNodeValue"
+    def script(xpath, visibility)
+      # this is done like so instead of a single pass over all elements due to a bug in Firefox:
+      # https://greasyfork.org/en/forum/discussion/12223/xpath-iteratenext-fails-in-firefox
+      xpath += visibility == 'hidden' ?
```

Comment:
Can we make the visibility value become a symbol everywhere ?

Reply:
Done.